### ISSUE TITLE
User flow when adding variant attributes can be improved

### ISSUE THREAD

**Issue by khalibloo** [mirumee/saleor#4619]
### What I'm trying to achieve
With the new enterprise grade attributes, when adding a variant attribute to a product type, the flow goes like so.
1. Check "This product type has variants"
2. Click "Save"
3. Click "Add Variant Attribute"
4. Maybe click "Save" again?

The main issue is with step 2. It's not clear to the user that they have to save before trying to add variant attribute. By convention, "save" is usually clicked after we've finished editing a form, not after each field.

### Describe a proposed solution
This is admittedly very tricky. Maybe propose to save the changes when user attempts to add variant attributes? Or even better, only add the attribute when "save" is clicked in the product type form.

### Other solutions I've tried and won't work
Automatically saving the product type without user consent can cause even more confusion for the user. For example, they might click "cancel" and expect that no modifications have been made to the product type.

**Comment by maarcingebala** [comment]
I agree that this UX is unintuitive. The reason why it works this way is that  "This product has variants" field represents the `ProductType.has_variants` field in the model. By default, this field is `False`. When creating a new product type you check it, add variant attributes and click "Save",  the mutation underneath fails to add variants attributes, as it's current state says that variants are disabled. 

I see two solutions:

1) Remove the `has_variants` field from the model and turn it into a property that would return `True` if any variant attributes are assigned to the given product type and `False` otherwise. Removing this field would result with an additional database query whenever we want to check if a product type has variants. This would require more changes in backend and we would have to think carefully whether we really need this field in the model.

2) Send two mutations when assigning the first variant attribute - one to update the product type and set `has_variants = True` and the other to actually assign the attribute.

I'm not sure which way is better so it would be nice to discuss it here.

**Comment by maarcingebala** [comment]
@dominik-zeglen What do you think about this? What that be a problem to implement the second solution and send two mutations in this case? This feels a bit hacky but I think it's less invasive than dropping a model field unless we'll agree that we don't need it.

**Comment by khalibloo** [comment]
@maarcingebala, the 2nd option will run into difficulties if the user decides to cancel in the end. Take this scenario for example.
1. Edit some fields
2. Add a variant attribute (edited fields are now saved)
3. Click cancel

The user will expect that his edits in step 1 will be undone.

**Comment by maarcingebala** [comment]
I've recorded a GIF to show the problem for those who aren't familiar:

![Kapture 2019-08-09 at 10 41 48](https://user-images.githubusercontent.com/5421321/62766424-bbc8c780-ba92-11e9-851c-49948417380a.gif)
